### ISSUE TITLE
Panic when `schemars` added as dependency in `Cargo.toml` file

### ISSUE THREAD

**Issue by rifatarefin** [verus-lang/verus-analyzer#33]
I have this content in `Cargo.toml` file:
```
[package]
name = "keyval"
version = "0.1.0"
edition = "2021"

[dependencies]
schemars = "0.8.21"

[package.metadata.verus.ide]
extra_args = "--crate-type=dylib --expand-errors"
```

The `schemars` dependency causes panic in the extension. I'm getting these pop-ups:
<img width="689" alt="Screenshot 2024-09-25 at 1 57 40 PM" src="https://github.com/user-attachments/assets/bb6a6bc0-c10c-490d-bc22-843a0a1cf7db">

Error log:

```
thread 'Worker' panicked at crates/parser/src/parser.rs:208:9:
assertion failed: self.eat(kind)
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic
   3: parser::grammar::verus::verus_ret_type
   4: parser::grammar::items::opt_item
   5: parser::grammar::items::item_or_macro
   6: parser::grammar::entry::top::source_file
   7: parser::TopEntryPoint::parse
   8: syntax::parsing::parse_text
   9: syntax::<impl syntax::ast::generated::nodes::SourceFile>::parse
  10: <base_db::ParseQuery as salsa::plumbing::QueryFunction>::execute
  11: salsa::Cycle::catch
  12: salsa::derived::slot::Slot<Q,MP>::execute
  13: salsa::derived::slot::Slot<Q,MP>::read
  14: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::fetch
  15: <DB as base_db::SourceDatabase>::parse
  16: hir_expand::db::parse_or_expand
  17: hir_def::item_tree::ItemTree::file_item_tree_query
  18: salsa::Cycle::catch
  19: salsa::derived::slot::Slot<Q,MP>::execute
  20: salsa::derived::slot::Slot<Q,MP>::read
  21: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::fetch
  22: <DB as hir_def::db::DefDatabase>::file_item_tree
  23: hir_def::nameres::collector::ModCollector::collect_module
  24: core::iter::adapters::copied::copy_fold::{{closure}}
  25: hir_def::nameres::collector::ModCollector::collect
  26: hir_def::nameres::collector::collect_defs
  27: hir_def::nameres::DefMap::crate_def_map_query
  28: salsa::Cycle::catch
  29: salsa::derived::slot::Slot<Q,MP>::execute
  30: salsa::derived::slot::Slot<Q,MP>::read
  31: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::fetch
  32: <DB as hir_def::db::DefDatabase>::crate_def_map
  33: hir_def::nameres::collector::DefCollector::import_macros_from_extern_crate
  34: hir_def::nameres::collector::ModCollector::process_macro_use_extern_crate
  35: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut
  36: hir_def::nameres::collector::ModCollector::collect
  37: hir_def::nameres::collector::collect_defs
  38: hir_def::nameres::DefMap::crate_def_map_query
  39: salsa::Cycle::catch
  40: salsa::derived::slot::Slot<Q,MP>::execute
  41: salsa::derived::slot::Slot<Q,MP>::read
  42: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::fetch
  43: <DB as hir_def::db::DefDatabase>::crate_def_map
  44: hir_def::nameres::path_resolution::<impl hir_def::nameres::DefMap>::resolve_path_fp_with_macro_single
  45: hir_def::nameres::path_resolution::<impl hir_def::nameres::DefMap>::resolve_path_fp_with_macro
  46: hir_def::nameres::attr_resolution::derive_macro_as_call_id
  47: alloc::vec::Vec<T,A>::retain::{{closure}}
  48: alloc::vec::Vec<T,A>::retain
  49: hir_def::nameres::collector::collect_defs
  50: hir_def::nameres::DefMap::crate_def_map_query
  51: salsa::Cycle::catch
  52: salsa::derived::slot::Slot<Q,MP>::execute
  53: salsa::derived::slot::Slot<Q,MP>::read
  54: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::fetch
  55: <DB as hir_def::db::DefDatabase>::crate_def_map
  56: hir::semantics::source_to_def::SourceToDefCtx::file_to_def
  57: hir::semantics::source_to_def::SourceToDefCtx::find_container
  58: hir::semantics::source_to_def::SourceToDefCtx::container_to_def
  59: hir::semantics::source_to_def::SourceToDefCtx::ancestors_with_macros
  60: hir::semantics::source_to_def::SourceToDefCtx::find_container
  61: hir::semantics::SemanticsImpl::with_ctx
  62: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::try_fold
  63: ide::hover::hover
  64: std::panic::catch_unwind
  65: rust_analyzer::handlers::request::handle_hover
  66: std::panic::catch_unwind
  67: core::ops::function::FnOnce::call_once{{vtable.shim}}
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[Error - 1:43:57 PM] Request textDocument/hover failed.
  Message: request handler panicked: assertion failed: self.eat(kind)
  Code: -32603 
Panic context:
> 
version: 0.0.0-standalone (324e554d8 2024-09-05)
request: textDocument/hover HoverParams {
    text_document: TextDocumentIdentifier {
        uri: Url {
            scheme: "file",
            cannot_be_a_base: false,
            username: "",
            password: None,
            host: None,
            port: None,
            path: "/Users/arefinmo/Downloads/Test%20Verus/keyval/src/key_types.rs",
            query: None,
            fragment: None,
        },
    },
    position: Position(
        Position {
            line: 52,
            character: 48,
        },
    ),
    work_done_progress_params: WorkDoneProgressParams {
        work_done_token: None,
    },
}
```

**Comment by rifatarefin** [comment]
Another scenario where it crashes:
`Cargo.toml` dependency:
```
[build-dependencies]
prost-build = "0.13.0"
```
<details>
<summary>Error log</summary>


```
thread 'PrimeCaches#7' panicked at crates/parser/src/parser.rs:208:9:
assertion failed: self.eat(kind)
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic
   3: parser::grammar::verus::verus_ret_type
   4: parser::grammar::items::opt_item
   5: parser::grammar::items::item_or_macro
   6: parser::grammar::entry::top::source_file
   7: parser::TopEntryPoint::parse
   8: syntax::parsing::parse_text
   9: syntax::<impl syntax::ast::generated::nodes::SourceFile>::parse
  10: <base_db::ParseQuery as salsa::plumbing::QueryFunction>::execute
  11: salsa::Cycle::catch
  12: salsa::derived::slot::Slot<Q,MP>::execute
  13: salsa::derived::slot::Slot<Q,MP>::read
  14: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::fetch
  15: <DB as base_db::SourceDatabase>::parse
  16: hir_expand::db::parse_or_expand
  17: hir_def::item_tree::ItemTree::file_item_tree_query
  18: salsa::Cycle::catch
  19: salsa::derived::slot::Slot<Q,MP>::execute
  20: salsa::derived::slot::Slot<Q,MP>::read
  21: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::fetch
  22: <DB as hir_def::db::DefDatabase>::file_item_tree
  23: hir_def::nameres::collector::ModCollector::collect_module
  24: core::iter::adapters::copied::copy_fold::{{closure}}
  25: hir_def::nameres::collector::ModCollector::collect
  26: hir_def::nameres::collector::ModCollector::collect_module
  27: core::iter::adapters::copied::copy_fold::{{closure}}
  28: hir_def::nameres::collector::ModCollector::collect
  29: hir_def::nameres::collector::collect_defs
  30: hir_def::nameres::DefMap::crate_def_map_query
  31: salsa::Cycle::catch
  32: salsa::derived::slot::Slot<Q,MP>::execute
  33: salsa::derived::slot::Slot<Q,MP>::read
  34: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::fetch
  35: <DB as hir_def::db::DefDatabase>::crate_def_map
  36: std::panic::catch_unwind
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</details>

**Comment by parno** [comment]
This should be fixed by [56d69f2](https://github.com/verus-lang/verus-analyzer/commit/56d69f2e4f02dc2bb27099fecd83c62fb2408661)
